src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -333,12 +333,15 @@ private IEdmVocabularyAnnotatable ComputeTarget()
                     container = this.schema.FindEntityContainer(targetSegments[0]);
                     if (container != null)
                     {
-                        IEdmEntityContainerElement containerElement = container.FindEntitySetExtended(targetSegments[1]);
+                        // Using the methods here results in a much faster lookup as it used a dictionary instead of using the list of container elements.

```

Comment:
```suggestion
                        // Using the methods here results in a much faster lookup as it uses a dictionary instead of using the list of container elements.

```